test/e2e/specs/site-editor/iframe-rendering.spec.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site editor iframe rendering mode', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Should render editor in standards mode.', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+
+		const compatMode = await page.evaluate(
+			() =>
+				document.querySelector( `iframe[name='editor-canvas']` )
+					.contentDocument.compatMode
+		);
```

Comment:
Nit: Let's change this into two steps:

```suggestion
		const compatMode = await page.locator( 'iframe[name="editor-canvas"]' )
			.evaluate( ( iframe ) => iframe.contentDocument.compatMode );
```